### PR TITLE
Join a workspace with a one-time control-center code

### PR DESCRIPTION
**Pull request by AstroQore**
## Summary

- Add `RemoteEnrollmentClient` (Core): mints a fresh P-256 signing keypair and a fresh P-256 key-agreement keypair, POSTs their x963 public halves plus the pairing code to `<control-url>/api/v1/enroll/consume`, and folds the 201 body into the exact `RemoteCoreProvisioning` a provisioning file produces. Default control URL `https://vibebar.aqor.io`; the origin is validated exactly like `relayURL` (bare HTTPS, no credentials/path/query/fragment).
- Install through the existing path: `RemoteCoreConfigStore.install(_ enrollment:)` stores the new Core identity in the credential Vault, then hands the provisioning to the same `install(_:)` the file import uses, so the Relay bearer token lands in the same Vault entry, `remote_core.json` is written by the same writer, and `RemoteProbeService.reconfigure()` picks the workspace up identically.
- Settings pane: when no remote configuration is installed, "Provisioning" now leads with a compact **Join with a code** affordance — code field (`VB-XXXXX-XXXXX`, trimmed and uppercased), Join button with inline progress, human-readable error text, and an optional "Use a different control center" URL override. The manual export/import flow is unchanged and still available underneath. After a successful join the pane refreshes to the connected state exactly like a file import.
- Error taxonomy surfaced to the UI, none of which can leak a secret: invalid/expired/consumed code (404 `invalid_code`), workspace already has a Core (409 `core_exists`), probe-role grant refused by a Core, rate limited (429), other HTTP status, transport failure, malformed 201 body, and an echoed Core key that does not match the keypair this Mac just generated.
- `RemoteCoreConfig` now accepts an empty probe map. A Mac that has just joined is the workspace's Core before any probe has enrolled; it authorizes no producer until one does, which is the correct fail-closed behavior.
- 12 new tests in `Tests/VibeBarCoreTests/RemoteEnrollmentTests.swift` run against a stubbed `URLProtocol` — no network, no Keychain, synthetic UUIDs and tokens only. The stub echoes the encryption key straight out of the request, so the happy path also asserts the request encoding.

### Follow-up (not in this PR)

A freshly joined Core has no registered probe signing keys, so it cannot yet open envelopes from probes enrolled afterwards. Learning those keys from the control API (`GET /api/v1/workspaces/:ws/devices`) is the natural next step; today the only way to register them is still a provisioning-file import.

## Test plan

- [x] `swift build`
- [x] `swift test` — 1089 tests, 0 failures
- [x] `./Scripts/build_app.sh release`
- [x] `codesign -d --entitlements - ".build/Vibe Bar.app"` → empty `<dict></dict>`, no `app-sandbox`
- [x] `codesign --verify --deep --strict ".build/Vibe Bar.app"` → clean
- [x] AGENTS.md § 6.3 home-directory grep → only the two pre-existing hits inside `RealHomeDirectory.swift`
- [ ] Manual smoke test of the join flow against a live control center — not run here: AQ's installed Vibe Bar was running, and launching a second instance from the worktree would have shared `~/.vibebar/` and the credential Vault with it.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: Sources/VibeBarCore/Services/RemoteEnrollmentClient.swift [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+        // malformed response rather than an unusable installed config.
+        do {
+            let provisioning = try RemoteCoreProvisioning(
+                workspaceID: workspaceID,
+                coreDeviceID: deviceID,
+                relayURL: relayURL,
+                relayBearerToken: bearerToken,
+                coreEpoch: epochValue,
+                ingestKeyID: keyID,
+                probeSigningPublicKeys: [:]
```
[the remark this answers is not in the document]
Comment (AstroQore) [reply Sources/VibeBarCore/Services/RemoteEnrollmentClient.swift:310]: Fixed in cb7dafb — the Core now learns its roster from the Relay instead of trusting whatever was installed once.

- `RemoteRelayClient.fetchDevices()` calls `GET /v1/workspaces/{workspace}/devices` with the same device-bearer credential, no-redirect session, and bounded response as the batch calls.
- `RemoteProbeRoster.resolve(from:)` derives the authorized-producer map from `role == "probe" && status == "active"` rows. Keys that are not 65-byte `0x04`-prefixed x963 points are skipped and counted, never trusted; the count is logged, the value never is. Revoked devices drop out by construction.
- `RemoteProbeService` resolves the roster once per sync cycle *before* opening any envelope, persists through `RemoteCoreConfigStore.updateProbeRoster` only when the map actually changed, and returns the installed map unchanged on any fetch failure — an unreachable or older Relay degrades to "no new probes", never to "reject everything".

This runs for both provisioning-file and code-joined configs, so it also fixes the staleness a file-provisioned Core hits after later enrollments or revocations.

The join success message no longer over-promises: "Syncing with the Relay; machines appear automatically as probes enroll."

New tests in `Tests/VibeBarCoreTests/RemoteDeviceRosterTests.swift`: active-only filtering, malformed keys skipped and counted, new probe added / revoked probe dropped, a code-joined Core learning its first probe, an unchanged roster producing no store write, plus roster parsing and rejection cases against a stubbed URLProtocol.

One cross-repo note for vibebar-relay: the roster response must carry `Cache-Control: no-store` like the other Relay reads — `RemoteRelayClient.validate` enforces it, and without the header the Core will simply keep its existing map.

File: Sources/VibeBarCore/Storage/RemoteCoreConfigStore.swift [hunk 2]
```diff
@@ -46,6 +46,15 @@ public enum RemoteCoreConfigStore {
         }
     }
 
+    /// Commit an enrollment obtained from the web control center. Identical to
+    /// the provisioning-file path once the identity is in place: the Core keys
+    /// were minted during the exchange rather than before it, so they have to
+    /// land in the Vault before the workspace binding that depends on them.
+    public static func install(_ enrollment: RemoteEnrollmentResult) throws {
+        try RemoteCoreIdentityStore.store(enrollment.identity)
+        try install(enrollment.provisioning)
```
[the remark this answers is not in the document]
Comment (AstroQore) [reply Sources/VibeBarCore/Storage/RemoteCoreConfigStore.swift]: Fixed in cb7dafb — the spent code is no longer a dead end.

- `install` is now explicitly idempotent and documented as such: the Vault replaces an entry with the same service/account, and the config file is written atomically, so re-running with the same `RemoteEnrollmentResult` converges from any partial state. The side effects moved behind an internal `RemoteCoreConfigSink` (`.live` is the only sink product code uses) purely so the retry path is testable without touching the real credential Vault or `~/.vibebar/`.
- The network exchange and the local save are now separate steps in `RemoteSettingsSection`. `saveEnrollment(_:)` is the only place the result reaches disk and the Keychain; when it throws, the `RemoteEnrollmentResult` is retained in `@State` and a **Retry saving** button re-runs *only* the local install — no new HTTP call, no new code. A **Discard** button sits next to it, and the retained result is cleared on success, on Discard, and on disconnect.
- The error text says explicitly that the code is already used and not to request another one. The bearer token is never rendered and never logged.

New test `testFailedInstallIsRecoverableWithTheSameEnrollment`: a failing config write surfaces (identity + token already landed, config absent), retrying with the same result produces a fully installed config, and a third install is still a safe no-op.

File: Sources/VibeBarApp/Views/RemoteSettingsSection.swift [hunk 3]
```diff
@@ -102,6 +121,131 @@ struct RemoteSettingsSection: View {
...
+            if pendingEnrollment != nil {
+                HStack(spacing: 10) {
+                    Button {
+                        retrySavingEnrollment()
+                    } label: {
+                        Label("Retry saving", systemImage: "arrow.clockwise")
+                    }
+                    Button("Discard") {
+                        pendingEnrollment = nil
+                        joinError = nil
```
[the remark this answers is not in the document]
Comment (AstroQore) [reply Sources/VibeBarApp/Views/RemoteSettingsSection.swift]: Fixed in 4f97f40 — the unfinished join now outlives the pane and the process.

- The consumed enrollment is written to the credential Vault **before** the first install attempt, under the same remote-sync service with a dedicated `pending-enrollment-v1` account. The record carries the provisioning (including the Relay bearer token), both Core private keys, and the device name. Nothing goes to disk outside the Vault, and nothing about it is logged.
- `RemoteCoreConfigStore.install` clears the record once the binding has committed (best-effort, after the commit). A record found while a workspace is already installed is stale by definition, so `pendingEnrollment()` returns nil and cleans it up instead of offering a resume.
- On `.onAppear` — only there, never in the background — the pane restores the record when no configuration is installed and offers **Resume saving**, with a caption explaining that the code is spent and a resume is the right move. In-session failures still read "Retry saving".
- **Discard** now goes through a confirmation dialog that names the server-side recovery: "Its pairing code is already spent, and the workspace still counts this Mac as its Core. To join again, revoke this Mac in the web control center, then create a fresh code." Discard and disconnect both delete the Vault record.
- Corner case covered: if the Vault write itself fails, `try?` lets the flow fall through to exactly the previous in-memory behavior — strictly no worse than before.

New tests in `RemoteEnrollmentTests`: `testPendingEnrollmentRecordRoundTripsIntact` (full round-trip through the real Vault payload encoder, both private key halves included, corrupt record fails closed), `testPendingEnrollmentSurvivesAFailedInstallAndResumesLater` (retain → failed install → record intact → resume completes → record cleared), `testInstalledWorkspaceClearsAStalePendingRecord`, and `testDiscardingAPendingEnrollmentDeletesTheRecord`. All run through the `RemoteCoreConfigSink` seam, so no real Keychain or `~/.vibebar/` writes.
